packages/flutter/lib/src/material/button.dart

```diff
@@ -46,6 +47,7 @@ class RawMaterialButton extends StatefulWidget {
     this.constraints = const BoxConstraints(minWidth: 88.0, minHeight: 36.0),
     this.shape = const RoundedRectangleBorder(),
     this.animationDuration = kThemeChangeDuration,
+    this.clipBehavior = Clip.none,
```

Comment:
all the constructors where you add clipBehavior should assert that it's not null, and should mention in their documentation that it must not be null.